File: src/OrbitShield.Application/Auth/AuthDtos.cs
namespace OrbitShield.Application.Auth;

public sealed record RegisterUserRequest(string Name, string Email, string Password, string Role);

public sealed record LoginRequest(string Email, string Password);

public sealed record AuthResponse(
    string AccessToken,
    string TokenType,
    DateTime ExpiresAt,
    UserProfileResponse User);

public sealed record UserProfileResponse(int Id, string Name, string Email, string Role);

File: android/OrbitShieldEngineer/app/src/main/java/com/orbitshield/engineer/data/remote/dto/AuthDtos.kt
package com.orbitshield.engineer.data.remote.dto

data class LoginRequestDto(
    val email: String,
    val password: String
)

data class AuthResponseDto(
    val accessToken: String,
    val tokenType: String,
    val expiresAt: String,
    val user: UserProfileDto
)

data class UserProfileDto(
    val id: Int,
    val name: String,
    val email: String,
    val role: String
)
